apps/src/StudioApp.js

```diff
@@ -1323,6 +1323,48 @@ StudioApp.prototype.onMouseMoveVizResizeBar = function (event) {
   this.resizeVisualization(newVizWidth);
 };
 
+StudioApp.prototype.clearVisualizationSize = function () {
+  if ($('#visualizationColumn').hasClass('wireframeShare')) {
+    return;
+  }
+
+  const editorColumn = $(".editor-column");
+  const visualization = document.getElementById('visualization');
+  const visualizationResizeBar = document.getElementById('visualizationResizeBar');
+  const visualizationColumn = document.getElementById('visualizationColumn');
+
+  if (this.isRtl()) {
+    visualizationResizeBar.style.right = '';
+    editorColumn.css('right', '');
+  } else {
+    visualizationResizeBar.style.left = '';
+    editorColumn.css('left', '');
+  }
+
+  visualizationResizeBar.style.lineHeight = '';
+  visualizationColumn.style.maxWidth = '';
+  visualization.style.maxWidth = '';
+  visualization.style.maxHeight = '';
+
+  if (!utils.browserSupportsCssMedia()) {
+    visualization.style.height = '';
+    visualization.style.width = '';
+  }
+
+  applyTransformScaleToChildren(visualization, '');
+
+  $('#soft-buttons').removeClass('soft-buttons-compact');
+
+  const smallFooter = document.querySelector('#page-small-footer .small-footer-base');
+  if (smallFooter) {
+    smallFooter.style.maxWidth = '';
+    const smallPrint = smallFooter.querySelector('small');
+    smallPrint.style.float = '';
+  }
+
+  utils.fireResizeEvent();
```

Comment:
It appears that the only css rules html2canvas has trouble with are the rules that have been rewritten by applab in `adjustAppSizeStyles`. html2canvas appears to see the original rules rather than the rewritten rules. 

By disabling the rewriting code, and touching up `applab/style.scss` to look the same as if it had been rewritten, I am able to capture screenshots with the correct dimension and scaling while maintaining the correct responsive behavior, without pinning/unpinning. This also allows capturing the entire image (or a center crop) instead of just the top 2/3rds. This path forward results in the simplest code, but it means we give up the ability to customize our responsive rules for future applab apps of other dimensions besides 320x450 (or we could add back the rule-rewriting and lose the ability to capture thumbnails of them).

The path of cloning the document and then pinning the viz column width of the clone seems doable in theory but also quite hacky -- it could be done by extracting a "document root" parameter out of `StudioApp.prototype.resizeVisualization` and avoiding any jquery inside. This seems hacky since it seems easy to introduce a change to `resizeVisualization` which changes how thumbnails are rendered.

Let's discuss options tomorrow, and consider deferring this until after the rest of the teacher view work as you suggested.